autorest/sender_test.go

```diff
@@ -902,3 +902,30 @@ func TestDoRetryForStatusCodes_NilResponseFatalError(t *testing.T) {
 		t.Fatalf("autorest: Sender#TestDoRetryForStatusCodes_NilResponseFatalError -- Got: nil error or wrong number of attempts - %v", err)
 	}
 }
+
+func TestDoRetryForStatusCodes_Cancel429(t *testing.T) {
+	retries := 6
+	client := mocks.NewSender()
+	resp := mocks.NewResponseWithStatus("429 Too many requests", http.StatusTooManyRequests)
+	client.AppendAndRepeatResponse(resp, retries)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(retries/2)*time.Second)
+	defer cancel()
+	req := mocks.NewRequest().WithContext(ctx)
+	r, err := SendWithSender(client, req,
+		DoRetryForStatusCodes(1, time.Duration(time.Second), http.StatusTooManyRequests),
+	)
+
```

Comment:
[Low-pri] I like to add a clause in my cancellation tests to make sure that cancellation is recognized and acted-on quickly.